72 take part page

davidtrussler opened the pull request:
[Trello](https://trello.com/c/isQwsOg9/72-take-part-page)

Ports the "Take part" landing page, update order page and delete page to the design system. Screenshots below. 

| Page | Design System |
|-|-|
| Landing page | ![Screenshot 2023-04-20 at 14 38 28](https://user-images.githubusercontent.com/6080548/233384821-81035b6d-dc98-4af8-bd01-5e2b488449ca.png) |
| Update order page | ![Screenshot 2023-04-19 at 12 38 43](https://user-images.githubusercontent.com/6080548/233063576-97c3e765-b16b-4e49-a65a-6916ef9819b2.png) |
| Delete page | ![Screenshot 2023-04-19 at 12 31 31](https://user-images.githubusercontent.com/6080548/233062052-37e1ca9b-f45e-4a5e-8bf9-66a10baab7d2.png) |

nnagewad commented:
Had a great chat with @davidtrussler.

Some minor changes:

- Update width to full as reorder component only works for full-width container
    - Will need to revisit the component itself and make necessary updates
- Make the "Update order" button primary
- Make "Add new take part page" a link and have it right next to the primary button

nnagewad commented:
as per conversation on slack, designs have been updated.

davidtrussler commented:
> 

@nnagewad I've updated the styles on the Edit/Delete links in the table to match the example you gave. That will be more consistent and looks better I think. I've updated the screenshot for the landing page above ☝️

nnagewad commented:
Looks great @davidtrussler

davidgisbey commented on app/views/admin/take_part_pages/index.html.erb:
```diff
@@ -1,38 +1,59 @@
-<div class="get-involved-header">
-  <h1>Get involved</h1>
-  <%= link_to "View on website", get_involved_url %>
+<div class="govuk-grid-row">
```
Does this need to live inside a row as it's full width anyway?

davidgisbey commented on app/views/admin/take_part_pages/index.html.erb:
```diff
@@ -1,38 +1,59 @@
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-full">
+    <p class="govuk-body">
+      <%= link_to "View on website", get_involved_url, class: "govuk-link", target: "_blank" %>
```
Does this need to live inside a row as it's full width anyway?

davidtrussler replied:
I guess strictly speaking it doesn't, but it seems to be the established pattern to use `govuk-grid-column-full` inside `govuk-grid-row` where we have full width elements, so I'd be inclined to leave it like this here as well.

davidgisbey commented on app/views/admin/take_part_pages/index.html.erb:
```diff
@@ -1,38 +1,59 @@
-  <p class="warning">
-    Warning: changes to take part page ordering appears instantly on the live site.
-  </p>
+<%= render "components/secondary_navigation", {
```
On other pages that use 2/3 width this is in the 2/3 column. I think we need to do that here

davidtrussler replied:
Yes, I think that's correct. Actually it isn't that way in the design for the page but it is more consistent, not least with the "Get involved" tab.

davidgisbey commented on app/controllers/admin/take_part_pages_controller.rb:
```diff
@@ -53,7 +62,7 @@ def reorder
   def get_layout
     design_system_actions = []
-    design_system_actions += %w[new create] if preview_design_system?(next_release: false)
+    design_system_actions += %w[new create index edit confirm_destroy update_order] if preview_design_system?(next_release: false)
```
I think edit might have snuck in here by accident  

Should it be
```
%w[new create index confirm_destroy update_order]

```

davidgisbey commented on app/views/admin/take_part_pages/index.html.erb:
```diff
@@ -1,38 +1,59 @@
+              },
+              {
+                text: (
+                  link_to("Edit", [:edit, :admin, take_part_page]) +
```
this is missing the `govuk-link` class

davidtrussler replied:
Yes, good point. And the delete link needs it as well in fact (for focus styles at least).

davidtrussler commented on app/views/admin/take_part_pages/index.html.erb:
```diff
@@ -1,38 +1,64 @@
+          items: @take_part_pages.map do |take_part_page|
+            {
+              id: take_part_page.id,
+              title: link_to(take_part_page.title, [:edit, :admin, take_part_page])
```
Yes, good point. And the delete link needs it as well in fact (for focus styles at least).

davidgisbey reviewed (approved):
lgtm!
